### PR TITLE
Package argp

### PR DESCRIPTION
**Pull request by davidbrochart**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by wolfv** [comment]
@davidbrochart I can debug on OSX locally if you want?

**Comment by davidbrochart** [comment]
Yes, that would be great!

**Comment by wolfv** [comment]
alright!

**Comment by davidbrochart** [comment]
Awesome @wolfv!

**Comment by wolfv** [comment]
With this we can have a working zchunk on OS X

I also had to use this endian.h polyfill: https://gist.github.com/yinyin/2027912

File: recipes/argp-standalone/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    - make
+
+test:
+  commands:
+    - test -f ${PREFIX}/lib/libargp.a
+    - test -f ${PREFIX}/include/argp.h
+
+about:
+    home: https://www.lysator.liu.se/~nisse/misc
+    license: GPL-2.0-or-later
```
Comment (wolfv) [inline recipes/argp-standalone/meta.yaml:30]: you included a license file for BSD3?

File: recipes/argp-standalone/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+{% set name = "argp" %}
```
Comment (wolfv) [inline recipes/argp-standalone/meta.yaml:1]: should we call it argp-standalone?

File: recipes/argp-standalone/meta.yaml (hunk 1 above)
Comment (davidbrochart) [reply recipes/argp-standalone/meta.yaml:30]: Yes I think I was confused when looking for packages with a GPLv2 license, the feedstock is BSD3.

File: recipes/argp-standalone/meta.yaml (hunk 2 above)
Comment (davidbrochart) [reply recipes/argp-standalone/meta.yaml:1]: Yes, I will change that, thanks.

**Comment by wolfv** [comment]
Actually I think the code for argp was taken from glibc which (I believe) is released under LGPL (https://en.wikipedia.org/wiki/GNU_C_Library)

This is the license from the argp.h file in this package:

```
/* Hierarchial argument parsing.
   Copyright (C) 1995, 96, 97, 98, 99, 2003 Free Software Foundation, Inc.
   This file is part of the GNU C Library.
   Written by Miles Bader <miles@gnu.ai.mit.edu>.

   The GNU C Library is free software; you can redistribute it and/or
   modify it under the terms of the GNU Library General Public License as
   published by the Free Software Foundation; either version 2 of the
   License, or (at your option) any later version.

   The GNU C Library is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
   Library General Public License for more details.

   You should have received a copy of the GNU Library General Public
   License along with the GNU C Library; see the file COPYING.LIB.  If not,
   write to the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
   Boston, MA 02111-1307, USA.  */
```

**Comment by wolfv** [comment]
the other tiny issue is that we're currently building a static library while a shared one would be preferred. I can try to fix that tomorrow.

**Comment by wolfv** [comment]
I tried to fix this by adding a CMakeLists but it wasn't easy.

**Pull request merged**